activetable.xml

```diff
@@ -1,118 +1,183 @@
+                    return False
+                orig_state = cell.get_state()
+                if not all(orig_state[k] == cell_state[k] for k in orig_state):
+                    return False
+            else:
+                if not str(cell) == cell_state:
+                    return False
+    return True
+
+
+def check_table(unused_expect, ans):
+    """Determine the correctness of all student responses.
+
+    The correctness is independently determined by the JavaScript code on the client side.
+    """
+    ans = json.loads(ans)
+    state = json.loads(ans['state'])
+
+    if not check_consistency(state):
+        return {
+            'ok': False,
+            'msg': 'The problem was edited by the course staff since the last time you '
+            'worked on it.  Please click the "Reset" button to get the latest version.'
+        }
+
+    correct_answers = total_questions = 0
+    for row, row_state in zip(table, state):
+        for cell, cell_state in zip(row, row_state):
+            if isinstance(cell, ResponseCell):
+                total_questions += 1
+                correct_answers += cell.check_response(cell_state['value'])
+
+    if correct_answers == total_questions:
+        return {'ok': True, 'msg': 'Great job!'}
     else:
-        return {'ok': False, 'msg': 'You have %s cells out of %s correct.' % (str(numCorrects),str(len(response.keys()))) }
+        return {
+            'ok': False,
+            'msg': 'You have {} cells out of {} correct.'.format(correct_answers, total_questions)
+        }
+
+## Problem definition ##
+## Edit the table and settings below. ##
+
+# The default tolerance in percent, used for all numeric fields that don't specify an ecplicit
+# tolerance.
+default_tolerance = 1.0
+
+# The table itself.  The first row contains the column headers.  Any cell in the following rows can
+# be either a string or number, which will simply be shown to the student, or one of the special
+# cell types Numeric() or String(), indicating a numeric or string input field that must be filled
+# in by the student.
+# Syntax of the special cell types:
+#     Numeric(answer=<correct_answer>, tolerance=<tolerance in percent>)
+#     String(answer='<correct answer>')
+table = [
+    ['Carts', 'v_initial', 'v_final', 'Delta v'],
+    ['green', 2.0, -1.2, Numeric(answer=-3.2, tolerance=2.0)],
+    ['red', -2.0, 1.2, Numeric(answer=3.2)],
+    ['yellow', -1.4, 0.0, Numeric(answer=1.4)],
+    ['purple', 3.4, 2.1, Numeric(answer=-1.3)],
+]
+
+# Initialisation -- don't remove!
+initial_state, height = init(table)
 
 ]]>
 </script>
 <p>
-Testing out an active table.
+  This is an ActiveTable problem, allowing you to place input fields in the cell
+  of a table.  You can edit the table and declare the cells that are input cells
+  by clicking EDIT above and following the instructions in the Python code.
+</p>
+<p>
+  It is recommended to set "Randomization" to "On Reset" and "Show Reset Button"
```

Comment:
@smarnach one more nit: it might make sense to mention that "Randomization" and "Show Reset Button" are *course* settings, accessible via "Settings" -> "Advanced Settings". It took me a bit of time to find them.